### DISCUSSION TITLE
Deploying a Next.js v11 app with `_middleware.js` to Vercel

### DISCUSSION THREAD

**Discussion by dstaley** [vercel/community#171] [Help] [answered]
I feel like the pieces are all there, but I'm not quite able to deploy a Next v11 app using `_middleware.js` to Vercel. I'm able to get things working locally using `vercel build && vercel dev`, but I'm not quite sure how to get Vercel to essentially perform the same steps that `vercel build` is doing.

Through some truly unholy configuration changes, I was able to run `vercel build` during the build process for my deployment, but I guess that doesn't configure the Next.js output in the way that Vercel would like, since it errors out with

```
TypeError: Cannot read property 'map' of undefined
at getWorkerConfig (/vercel/e884033519854e23/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:671729)
at async Promise.all (index 0)
at getWorkerSource (/vercel/e884033519854e23/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:670725)
at Object.serverBuild (/vercel/e884033519854e23/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:638843)
at buildStep (/var/task/webpack:/@vercel/api/scripts/build-container/src/utils/purpose-build.ts:229:1)
```

Does anyone have any tips? I know Next.js v12 natively supports this, but it's clear that things _can_ work on v11; I'm sure I'm just missing the magic configuration combinations to get everything to work.

**Comment by leerob** [comment, marked as answer]
Middleware was released with Next.js 12: http://nextjs.org/12

I would recommend updating to v12 instead of trying to use Middleware without v12.

**Reply by dstaley** [reply]
Oh absolutely! However I was hoping that since Vercel Edge Functions are framework agnostic, I'd be able to at least use the root-level _middleware.js method.

Is there something about the way Vercel builds/runs v11 apps that makes it difficult to use Edge Functions in the way they can be used in other frameworks? I know you've mentioned that there's some improvements coming down the pipeline for non-Next.js projects; maybe those would help?
